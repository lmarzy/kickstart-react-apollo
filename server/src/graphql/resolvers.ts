import { movieResolver, spaceXResolvers } from '../modules';

export const resolvers = [movieResolver, spaceXResolvers];
