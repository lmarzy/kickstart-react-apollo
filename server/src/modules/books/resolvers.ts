import { booksData, booksDataInterface } from './data';

export const bookResolver = {
  Query: {
    books: (): booksDataInterface[] => booksData,
  },
};
