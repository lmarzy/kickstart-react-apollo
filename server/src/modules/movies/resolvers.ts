import { moviesData, moviesDataInterface } from './data';

export const movieResolver = {
  Query: {
    movies: (): moviesDataInterface[] => moviesData,
  },
};
