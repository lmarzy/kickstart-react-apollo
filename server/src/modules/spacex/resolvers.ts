import { SpaceXData } from './data';

export const spaceXResolvers = {
  Query: {
    rockets: (
      _root: null,
      _args: null,
      { dataSources }: any,
    ): Promise<SpaceXData[]> => dataSources.spaceXDatasource.rockets(),
  },
};
