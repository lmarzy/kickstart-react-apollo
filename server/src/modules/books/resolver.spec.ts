import { bookResolver } from './resolvers';

describe('books resolver', async () => {
  test('full response', async () => {
    const actualResult = await bookResolver.Query.books();

    expect(actualResult).toMatchSnapshot();
  });
});
