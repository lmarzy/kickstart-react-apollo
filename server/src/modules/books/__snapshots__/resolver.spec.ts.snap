// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`books resolver full response 1`] = `
Array [
  Object {
    "author": "Harper Lee",
    "id": 1000,
    "title": "To Kill a Mockingbird",
  },
  Object {
    "author": "F.Scott Fitzgerald",
    "id": 1001,
    "title": "The Great Gatsby",
  },
]
`;
